relay-quotas/src/redis.rs

```diff
@@ -383,6 +385,73 @@ mod tests {
         );
     }
 
+    /// Tests that a quota with and without namespace are counted separately.
+    #[test]
+    fn test_non_global_namespace_quota() {
+        let get_quota = |namespace: Option<MetricNamespace>| -> Quota {
+            Quota {
+                id: Some(format!("test_simple_quota_{}", uuid::Uuid::new_v4())),
+                categories: DataCategories::new(),
+                scope: QuotaScope::Organization,
+                scope_id: None,
+                limit: Some(5),
+                window: Some(60),
+                reason_code: Some(ReasonCode::new(format!("ns: {:?}", namespace))),
+                namespace,
+            }
+        };
+
+        let quotas = &[get_quota(None)];
+        let quota_with_namespace = &[get_quota(Some(MetricNamespace::Transactions))];
+
+        let scoping = ItemScoping {
+            category: DataCategory::Error,
+            scoping: &Scoping {
+                organization_id: 42,
+                project_id: ProjectId::new(43),
+                project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+                key_id: Some(44),
+            },
+            namespace: Some(MetricNamespace::Transactions),
+        };
+
+        let rate_limiter = build_rate_limiter();
+
+        for i in 0..10 {
```

Comment:
You're effectively having two loops in one, and splitting it increases readability:

```rs
for i in 0..5 {
  assert_eq!(
    rate_limits[0].reason_code,
    Some(ReasonCode::new("ns: None"))
  );
}
for i in 0..5 {
  assert_eq!(rate_limits, vec![]);
}
```
(Same with the loop below)